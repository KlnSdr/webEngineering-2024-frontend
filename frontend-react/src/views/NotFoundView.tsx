function NotFoundView() {
  return <div>
    <h1>ups!</h1>
    <h1>diese url kennen wir nicht</h1>
  </div>;
}

export default NotFoundView;
