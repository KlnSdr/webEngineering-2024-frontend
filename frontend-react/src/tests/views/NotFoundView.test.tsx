import React from "react";
import { render } from "@testing-library/react";
import NotFoundView from "../../views/NotFoundView";

describe("NotFoundView Component", () => {

    it('matches snapshot for NotFoundView component', () => {
        const { asFragment } = render(<NotFoundView />);
        expect(asFragment()).toMatchSnapshot();
    });

});
