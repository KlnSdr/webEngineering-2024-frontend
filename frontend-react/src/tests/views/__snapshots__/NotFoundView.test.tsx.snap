// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundView Component matches snapshot for NotFoundView component alone 1`] = `
<DocumentFragment>
  <div>
    <h1>
      ups!
    </h1>
    <h1>
      diese url kennen wir nicht
    </h1>
  </div>
</DocumentFragment>
`;

exports[`NotFoundView Component renders 404 NotFoundView for unknown routes and matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top navbar-inverse"
  >
    <button
      aria-controls="navbarNav"
      aria-expanded="false"
      aria-label="Toggle navigation"
      class="navbar-toggler"
      data-bs-target="#navbarNav"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse"
      id="navbarNav"
    >
      <ul
        class="navbar-nav"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/recipe/new"
          >
            neues Rezept
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/search"
          >
            Suche
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/survey"
          >
            Umfrage
          </a>
        </li>
      </ul>
      <ul
        class="navbar-nav ms-auto"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/login"
          >
            Anmelden
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    class="App"
  >
    <div>
      <h1>
        ups!
      </h1>
      <h1>
        diese url kennen wir nicht
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
